central/complianceoperator/v2/report/manager/format/formatter_test.go

```diff
@@ -27,62 +33,139 @@ type ComplianceReportingFormatterSuite struct {
 }
 
 func (s *ComplianceReportingFormatterSuite) Test_FormatCSVReportNoError() {
-	matcher := &valueMatcher{
-		data: getFakeReportData(),
-	}
-	s.zipWriter.EXPECT().Create(gomock.Any()).Times(1).Return(nil, nil)
-	s.csvWriter.EXPECT().AddValue(matcher).Times(2).Do(func(_ any) {
-		matcher.recordNumber++
+	s.Run("with empty failed clusters", func() {
+		s.zipWriter.EXPECT().Create(fmt.Sprintf(successfulClusterFmt, clusterID1)).Times(1).Return(nil, nil)
+		gomock.InOrder(
+			s.csvWriter.EXPECT().AddValue(gomock.Cond[csv.Value](func(target csv.Value) bool {
+				data := getFakeReportData()
+				return compareStringSlice(target, generateRecord(data[clusterID1][0]))
```

Comment:
nitpick: one consideration. Can we use assertion functions here? i.e. `s.Equal()` - it returns `bool` anyway, but what is important, logged information about differences is detailed, and it would be helpful to find issues in failed tests.